File: VB/WindowsFormsApplication475/CustomEdit.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraEditors
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq
Imports System.Text

Namespace WindowsFormsApplication475
	<ToolboxItem(True)> _
	Public Class CustomEdit
		Inherits ColorPickEdit

		'The static constructor that calls the registration method 
		Shared Sub New()
			RepositoryItemCustomEdit.RegisterCustomEdit()
		End Sub

		'Initialize the new instance 
		Public Sub New()
			'... 
		End Sub

		'Return the unique name 
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return RepositoryItemCustomEdit.CustomEditName
			End Get
		End Property

		'Override the Properties property 
		'Simply type-cast the object to the custom repository item type 
		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
		Public Shadows ReadOnly Property Properties() As RepositoryItemCustomEdit
			Get
				Return TryCast(MyBase.Properties, RepositoryItemCustomEdit)
			End Get
		End Property

	End Class

End Namespace

File: CS/WindowsFormsApplication475/CustomEdit.cs
using DevExpress.XtraEditors;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;

namespace WindowsFormsApplication475 {
    [ToolboxItem(true)]
    public class CustomEdit : ColorPickEdit {

        //The static constructor that calls the registration method 
        static CustomEdit() { RepositoryItemCustomEdit.RegisterCustomEdit(); }

        //Initialize the new instance 
        public CustomEdit() {
            //... 
        }

        //Return the unique name 
        public override string EditorTypeName {
            get {
                return
                    RepositoryItemCustomEdit.CustomEditName;
            }
        }

        //Override the Properties property 
        //Simply type-cast the object to the custom repository item type 
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemCustomEdit Properties {
            get { return base.Properties as RepositoryItemCustomEdit; }
        }

    }

}
